spec/system/user/user_show_spec.rb

```diff
@@ -0,0 +1,99 @@
+                                               bio: 'Back End Web Developer',
+                                               post_counter: 0
+                                             }
+                                           ])
+
+      @posts = @user1.posts.create([
+                                     {
+                                       text: 'My first post',
+                                       title: 'First post',
+                                       likes_counter: 0,
+                                       comments_counter: 0
+                                     },
+                                     {
+                                       text: 'My second post',
+                                       title: 'second post',
+                                       likes_counter: 0,
+                                       comments_counter: 0
+                                     },
+                                     {
+                                       text: 'My third post',
+                                       title: 'third post',
+                                       likes_counter: 0,
+                                       comments_counter: 0
+                                     },
+                                     {
+                                       text: 'My fourth post',
+                                       title: 'fourth post',
+                                       likes_counter: 0,
+                                       comments_counter: 0
+                                     }
+                                   ])
+    end
+
+    it 'I can see the profile picture for each user.' do
+      visit '/users'
+      expect(page).to have_content('photo')
+    end
+
+    it "I can see the user's username." do
+      visit user_path(@user1)
+      expect(page).to have_content('Charles')
+    end
+
+    it 'I can see the number of posts the user has written.' do
+      visit user_path(@user1)
+      expect(page).to have_content('Number of Posts: 4')
+    end
+
+    it "I can see the user's bio." do
+      visit user_path(@user1)
+      expect(page).to have_content('Full Stack Web Developer')
+    end
+
+    it "I can see a button that lets me view all of a user's posts." do
+      visit user_path(@user1)
+      page.should have_button('See all posts')
+    end
+
+    it "When I click to see all posts, it redirects me to the user's post's index page" do
+      visit user_path(@user1)
+      click_button('See all posts')
+      expect(page).to have_content('My first post')
+      expect(page).to have_content('My second post')
+      expect(page).to have_content('My third post')
+      expect(page).to have_content('My fourth post')
+    end
+
+    it "When I click a user's post, it redirects me to that post's show page." do
+      visit user_path(@user1)
+      click_link("post_#{@posts[0].id}_link")
+      page.should have_button('Add comments')
+    end
+  end
```

Comment:
- [ ] You did a great job working on this milestone! 💯 However, there's a missing test case:

![image](https://github.com/forLoop94/blogApp/assets/77069011/5b8de017-9a2e-4e62-8865-c36be3f66166)

Please consider adding it! 🤓 🔧